src/include/daos_fs_sys.h

```diff
@@ -0,0 +1,581 @@
+		return rc;
+	}
+
+	/**
+	 * Handle the case of root "/".
+	 * TODO double check this
+	 */
+	if (sys_path->name == NULL) {
+		sys_path->parent = NULL; /* TODO set to root explicitly? */
+		sys_path->name = sys_path->dir_name;
+		sys_path->dir_name = NULL;
+	}
+
+	return rc;
+}
+
+/**
+ * The calls below here should be as similar as possible to the
+ * POSIX versions, but with a dfs_sys_t parameter.
+ *
+ * TODO Do we want these to set errno and return -1, like POSIX,
+ * or do we want them to return errno, like DFS?
+ */
+int
+dfs_sys_access(dfs_sys_t *dfs_sys, const char* path, int amode)
+{
+	int		rc;
+	dfs_sys_path_t	sys_path;
+
+	rc = dfs_sys_path_init(dfs_sys, &sys_path, path);
+	if (rc)
+		D_GOTO(out, rc);
+
+	rc = dfs_access(dfs_sys->dfs, sys_path.parent, sys_path.name, amode);
+	if (rc) {
+		/* TODO do we want to log errors for the dfs_ calls? */
+		D_ERROR("dfs_access %s failed\n", sys_path.name);
+		D_GOTO(out_free_path, rc);
+	}
+
+out_free_path:
+	dfs_sys_path_free(&sys_path);
+out:
+	/* If we want to set errno, we can do this */
+	// errno = rc;
+	// rc = errno ? 1 : 0;
+	return rc;
+}
+
+int
+dfs_sys_faccessat(dfs_sys_t *dfs_sys, int dirfd, const char* path, int amode,
+		  int flags)
+{
+	int		rc;
+	dfs_sys_path_t  sys_path;
+	dfs_obj_t	*obj;
+	mode_t		mode;
+	int		lookup_flags = O_RDWR;
+
+	if (dirfd != AT_FDCWD)
+		D_GOTO(out, rc=ENOTSUP);
+
+	if (flags & AT_EACCESS)
+		D_GOTO(out, rc=ENOTSUP);
+
+	rc = dfs_sys_path_init(dfs_sys, &sys_path, path);
+	if (rc)
+		D_GOTO(out, rc);
+
+	if (flags & AT_SYMLINK_NOFOLLOW)
+		lookup_flags |= O_NOFOLLOW;
+
+	/* Lookup the obj and get it's mode */
+	rc = dfs_lookup_rel(dfs_sys->dfs, sys_path.parent, sys_path.name,
+			    lookup_flags, &obj, &mode, NULL);
+	if (rc) {
+		/* TODO do we want to log errors for the dfs_ calls? */
+		D_ERROR("dfs_lookup_rel %s failed\n", sys_path.name);
+		D_GOTO(out_free_path, rc);
+	}
+
+	/* A link itself is always accessible */
+	if (S_ISLNK(mode))
+		D_GOTO(out_free_obj, rc);
+
+	rc = dfs_access(dfs_sys->dfs, sys_path.parent, sys_path.name, amode);
+	if (rc) {
+		/* TODO do we want to log errors for the dfs_ calls? */
+		D_ERROR("dfs_access %s failed\n", sys_path.name);
+		D_GOTO(out_free_obj, rc);
+	}
+
+out_free_obj:
+	dfs_release(obj);
+out_free_path:
+	dfs_sys_path_free(&sys_path);
+out:
+	return rc;
+}
+
+int
+dfs_sys_chmod(dfs_sys_t *dfs_sys, const char* path, mode_t mode);
+
+int
+dfs_sys_utimensat(dfs_sys_t *dfs_sys, const char *pathname,
+		  const struct timespec times[2], int flags);
+
+int
+dfs_sys_lstat(dfs_sys_t *dfs_sys, const char* path, struct stat* buf);
+
+int
+dfs_sys_stat(dfs_sys_t *dfs_sys, const char* path, struct stat* buf);
+
+int
+dfs_sys_mknod(dfs_sys_t *dfs_sys, const char* path, mode_t mode, dev_t dev);
+
+/**
+ * TODO - Many more functions:
+ * mknod
+ * listxattr
+ * llistxattr
+ * getxattr
+ * lgetxattr
+ * lsetxattr
+ * readlink
+ * symlink
+ * open
+ * close
+ * lseek
+ * read
+ * pread
+ * write
+ * pwrite
```

Comment:
Will initially only have `pread` and `pwrite`. Caller will be responsible for offsets.